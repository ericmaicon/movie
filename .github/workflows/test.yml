name: Lint & Test

on: push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
          node-version: '14'
      - name: Cache dependencies
        uses: actions/cache@v2.1.4
        id: cache-dependencies
        with:
          path: |
            node_modules
            */*/node_modules
          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
      - name: Install Dependencies
        if: steps.cache-dependencies.outputs.cache-hit != 'true'
        run: yarn install && yarn bootstrap
      - name: Link
        run: yarn lerna link
      - name: Build deps
        run: yarn build
      - name: Lint
        run: yarn lint
      - name: Run Tests
        run: yarn test
      - uses: codecov/codecov-action@v1
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: packages/**/coverage
          fail_ci_if_error: true
