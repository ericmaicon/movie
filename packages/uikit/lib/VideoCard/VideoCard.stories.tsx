import React from 'react';

import { Story, Meta } from '@storybook/react';

import VideoCard, { VideoCardProps } from './VideoCard';

export default {
  title: 'Layout/VideoCard',
  component: VideoCard,
} as Meta;

const Template: Story<VideoCardProps> = (args) => <VideoCard {...args} />;

export const Default = Template.bind({});
Default.args = {
  src:
    'https://www.themoviedb.org/t/p/w220_and_h330_face/drulhSX7P5TQlEMQZ3JoXKSDEfz.jpg',
  alt:
    'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
};
