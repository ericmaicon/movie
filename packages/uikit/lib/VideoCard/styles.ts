import styled from 'styled-components';

export const VideoCard = styled.img`
  cursor: pointer;
  width: 100%;

  &:hover {
    box-shadow: 0px 0px 0px 2px rgba(0, 0, 0, 0.2);
  }
`;
