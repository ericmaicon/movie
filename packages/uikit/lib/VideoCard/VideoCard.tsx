import React, { ImgHTMLAttributes, useState } from 'react';

import * as Styled from './styles';

export interface VideoCardProps extends ImgHTMLAttributes<HTMLImageElement> {}

const VideoCard = ({
  src = 'https://i.stack.imgur.com/l60Hf.png',
  ...rest
}: VideoCardProps) => {
  const [source, setSource] = useState(src);

  function handleError() {
    setSource('https://i.stack.imgur.com/l60Hf.png');
  }

  return <Styled.VideoCard src={source} onError={handleError} {...rest} />;
};

export default VideoCard;
