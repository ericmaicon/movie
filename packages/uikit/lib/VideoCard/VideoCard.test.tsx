import React from 'react';

import { render } from '~/test/index';

import VideoCard from './VideoCard';

describe('VideoCard', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(
      <VideoCard
        data-testid="video-card"
        src="https://www.themoviedb.org/t/p/w220_and_h330_face/drulhSX7P5TQlEMQZ3JoXKSDEfz.jpg"
      />,
    );

    const textElement = getByTestId('video-card');

    expect(textElement).toBeInTheDocument();
    expect(textElement).toHaveAttribute('src');
    expect(textElement).toMatchSnapshot();
  });

  it('should render error image', () => {
    const { getByTestId } = render(<VideoCard data-testid="video-card" />);

    const textElement = getByTestId('video-card');

    expect(textElement).toMatchSnapshot();
  });

  it('should render error image', () => {
    const { getByTestId } = render(
      <VideoCard data-testid="video-card" src="some invalid url" />,
    );

    const textElement = getByTestId('video-card');

    expect(textElement).toMatchSnapshot();
  });
});
