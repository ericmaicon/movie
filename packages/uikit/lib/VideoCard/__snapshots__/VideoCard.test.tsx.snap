// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard should render correctly 1`] = `
.c0 {
  cursor: pointer;
  width: 100%;
}

.c0:hover {
  box-shadow: 0px 0px 0px 2px rgba(0,0,0,0.2);
}

<img
  class="c0"
  data-testid="video-card"
  src="https://www.themoviedb.org/t/p/w220_and_h330_face/drulhSX7P5TQlEMQZ3JoXKSDEfz.jpg"
/>
`;
